fix: fixed the bug on 'My Squads' section of the sidebar, to display the titles for the list of squads using displayName, instead of the current metadata name

03DhruvJoshi opened the pull request:
## Hey, I just made a Pull Request!

**Problem**: When accessing the squads that a user is part of from the sidebar, the metadata name gets displayed for each squad. They should instead use the spec.profile.displayName - This approach makes it easier for the logged in user to switch between squads as the actual title of the group will be displayed in the sidebar instead of the metadata.

**Solution**: Fixed this bug from the component function, where in 'My Squads' sub menu items section to display a list of Squads - I have used the EntityDisplayName function for the list of squads to display their title using spec.profile.displayName - instead of the current metadata name/title. This is to ensure increased readability to the user. In case there is no such property for displayName defined on the entities, then the function will display the metadata name. 

Current Version 
![Screenshot 2025-01-06 113704](https://github.com/user-attachments/assets/1477191c-84b6-4f4f-a90a-fd400a9d85b7)

Upcoming Version
![Screenshot 2025-01-06 113628](https://github.com/user-attachments/assets/d8a2d70f-b94c-4248-bb3e-e5dff54f92c3)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

fixes Github Issue #28363

awanlin commented on .changeset/purple-years-cross.md:
```diff
@@ -0,0 +1,5 @@
+'@backstage/plugin-org': patch
+---
+
+Fixed bug in MyGroupsSidebarItem component, for the 'My Squads' section to display a list of Squads using the displayName from the entity config, instead of using the metadata name and title. This is to ensure increased readability to the user.
```
Using backticks (`) for code will also help resolve the Vale errors 👍

awanlin commented on plugins/org/src/components/MyGroupsSidebarItem/MyGroupsSidebarItem.tsx:
```diff
@@ -96,7 +97,7 @@ export const MyGroupsSidebarItem = (props: {
           return (
             <SidebarSubmenuItem
-              title={group.metadata.title || group.metadata.name}
+              title={<EntityDisplayName entityRef={group} />}
```
As we have some tests for `MyGroupsSidebarItem` already can we please add some more specifically for these cases just so we don't have regressions now and in the future, please? Ideally these should be new ones for the checks on `name`, `title`, and `displayName`.

awanlin reviewed (changes requested):
Thanks for taking this on @03DhruvJoshi, left some small comments 👍

jwnmulder commented:
@03DhruvJoshi, looking at the added to-be screenshot, the icons are shown twice for each group

03DhruvJoshi commented:
> @03DhruvJoshi, looking at the added to-be screenshot, the icons are shown twice for each group

Great observation @jwnmulder, I have added set the hideIcon property in EntityDisplayName as true, fixed now: 

![image](https://github.com/user-attachments/assets/e2bc985e-64ca-4c2e-9ef1-b1d9e1350732)
